### DISCUSSION TITLE
Progress bar in training

### DISCUSSION THREAD

**Discussion by AzmiHaider92** [Lightning-AI/pytorch-lightning#19120] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi,

This is an example of a progress bar training on a dataset with 723 samples and batch size of 16:
Epoch 2:  49%|████▉     | 22/45 [00:26<00:27,  1.18s/it, loss=0.0981, v_num=m49m]

The number 45 is the number of batches in one epoch (723/16=45) 

However, sometimes when I run the same code it shows 723 instead of 45. 
Why is that? it's the same code with the same resources. 

Thanks in advance
